### PR TITLE
Fix setting of RANSAC max error in pose prior BA alignment

### PR DESCRIPTION
**Pull request by ahojnnes**
Fixes: https://github.com/colmap/colmap/issues/2992

### CODE REVIEW

**Review by B1ueber2y — APPROVED** [review]

File: src/colmap/estimators/bundle_adjustment.cc [hunk 1]
```diff
@@ -1011,16 +1011,21 @@ class PosePriorBundleAdjuster : public BundleAdjuster {
...
+          ++num_covs;
+        }
+      }
+      if (num_covs == 0) {
+        LOG(WARNING) << "No pose priors with valid covariance found.";
+        return false;
       }
-    }
-    if (num_covs > 0) {
       ransac_options.max_error = (3. * (avg_cov / num_covs).cwiseSqrt()).norm();
```
Comment (B1ueber2y) [inline src/colmap/estimators/bundle_adjustment.cc:1029]: Niche and not directly relevant to this PR: cWiseSqrt() + norm() should be numerically inferior to the square root of the sum?

Comment (ahojnnes) [reply src/colmap/estimators/bundle_adjustment.cc:1029]: Not sure I understand. sqrt(sum(...)) would not produce the same result?

Comment (B1ueber2y) [reply src/colmap/estimators/bundle_adjustment.cc:1029]: vec.cwiseSqrt().norm() and sqrt(vec.sum()) are mathematically equivalent?

Comment (ahojnnes) [reply src/colmap/estimators/bundle_adjustment.cc:1029]: Actually, looking at this more now, I would have expected the multiplier 3 to be outside the norm? I guess it is meant as the confidence interval, where we set the max error as 3 * stddev?

Comment (ahojnnes) [reply src/colmap/estimators/bundle_adjustment.cc:1029]: I am fully confused now. That's also the same. Excuse the brain fart :-)

**Review by sarlinpe — APPROVED** [review]

File: src/colmap/estimators/bundle_adjustment.cc [hunk 2]
```diff
@@ -1011,17 +1011,28 @@ class PosePriorBundleAdjuster : public BundleAdjuster {
...
-        ++num_covs;
+    if (prior_options_.ransac_max_error > 0) {
+      ransac_options.max_error = prior_options_.ransac_max_error;
+    } else {
+      double max_stddev_sum = 0;
+      size_t num_valid_covs = 0;
+      for (const auto& [_, pose_prior] : pose_priors_) {
+        if (pose_prior.IsCovarianceValid()) {
+          const double max_stddev =
+              std::sqrt(Eigen::SelfAdjointEigenSolver<Eigen::Matrix3d>(
```
Comment (ahojnnes) [inline src/colmap/estimators/bundle_adjustment.cc:1022]: @B1ueber2y I think this makes more sense now. PTAL. Thanks.

**Pull request merged**